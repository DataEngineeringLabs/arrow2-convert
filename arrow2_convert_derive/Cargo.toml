[package]
name = "arrow2_convert_derive"
version = "0.1.0"
authors = [
    "Jorge Leitao <jorgecarleitao@gmail.com>",
    "Chandra Penke <chandrapenke@gmail.com>"
]
edition = "2021"
license = "Apache-2.0 OR MIT"
keywords = ["Arrow", "arrow2"]
repository = "https://github.com/DataEngineeringLabs/arrow2-derive/arrow2_convert_derive"
description = "Proc macros for arrow2_convert"

[lib]
proc-macro = true

[dependencies]
syn = { version = "1", features=["full"] }
quote = "1"
proc-macro2 = "1"
proc-macro-error = { version = "1" }
