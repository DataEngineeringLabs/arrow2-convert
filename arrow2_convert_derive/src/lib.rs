use proc_macro2::TokenStream;
use proc_macro_error::proc_macro_error;
use quote::TokenStreamExt;

mod _struct;
mod input;

/// Derive macro for the Array trait.
#[proc_macro_error]
#[proc_macro_derive(ArrowField, attributes(arrow2_convert))]
pub fn arrow2_convert_derive_field(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
    let ast = syn::parse(input).unwrap();
    let input = input::Input::new(ast);

    // Build the output, possibly using quasi-quotation
    let mut generated = TokenStream::new();
    generated.append_all(_struct::expand_derive(&input));
    generated.into()
}
