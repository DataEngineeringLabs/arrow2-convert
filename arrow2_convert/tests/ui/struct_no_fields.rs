use arrow2_convert_derive::ArrowField;

#[derive(ArrowField)]
struct S {}

fn main() 
{}