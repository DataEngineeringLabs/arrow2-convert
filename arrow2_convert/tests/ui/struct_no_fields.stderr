error: Expected struct to have more than one field
 --> tests/ui/struct_no_fields.rs:4:8
  |
4 | struct S {}
  |        ^
